src/components/Reports/PartnerGivingAnalysisReport/PartnerGivingAnalysisReport.tsx

```diff
@@ -106,28 +122,28 @@ export const PartnerGivingAnalysisReport: React.FC<Props> = ({
     isRowChecked,
   } = useMassSelection(allContactIds);
 
-  const localeText = useTablePaginationLocaleText();
-
-  const handleRequestSort = (
-    _event: React.MouseEvent<unknown>,
-    property: string,
-  ) => {
-    const isAsc = orderBy === property && order === 'asc';
-    setOrder(isAsc ? 'desc' : 'asc');
-    setOrderBy(property as keyof Contact);
+  const handlePageChange = (model: { page: number; pageSize: number }) => {
+    if (model.pageSize !== paginationModel.pageSize) {
+      cursorsRef.current = new Map([[0, null]]);
+      setPaginationModel({ page: 0, pageSize: model.pageSize });
+    } else {
+      setPaginationModel(model);
+    }
   };
 
-  const handlePageChange = (
-    _event: React.MouseEvent<unknown> | null,
-    newPage: number,
-  ): void => {
-    setPage(newPage);
-  };
+  const handleSortChange = (model: GridSortModel) => {
+    const property = model[0].field;
+    const transformOrderBy = EnumMap[orderBy];
+    const isAsc = transformOrderBy === property && order === 'asc';
+    const newOrder = isAsc ? 'desc' : 'asc';
```

Comment:
What is the `transformOrderBy === property` check for? Would this work?

```ts
const newOrder = order === 'asc' ? 'desc' : 'asc';
```

But more fundamentally, wouldn't `model[0].sort` tell you the new order?